python/lsst/ctrl/stats/databaseManager.py

```diff
@@ -20,36 +20,71 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
-from lsst.cat.MySQLBase import MySQLBase
+from lsst.db import utils
+from lsst.db.engineFactory import getEngineFromArgs
+# from lsst.db.testHelper import readCredentialFile
 
 
-class DatabaseManager(MySQLBase):
-    """Convenience class the MySQLBase class with which we hold the
-    the username and password.  We do this so we can pass this object
-    around and not have to pass the user name and password all over
-    the place.  Note: It was pointed out that this functionality may be
-    better placed in MySQLBase itself, and if that happens this code
-    should be removed and that new code should be used instead.
+class DatabaseManager(object):
+    """
     """
 
-    def __init__(self, dbHostName, portNumber, user, password):
-        """Creates a connection to a MySQL server
-        @param dbHostName: a server where the MySQL daemon resides
-        @param portNumber: the port number the MySQL daemon is listening on
+    def __init__(self, dbHostName, portNumber, user, passwd):
+        """Creates a connection to an sql server
+        @param dbHostName: a server where the database daemon resides
+        @param portNumber: the port number the database daemon is listening on
         @param user: the user name to connect as
-        @param password: the users's password
+        @param passwd: the users's password
         """
-        MySQLBase.__init__(self, dbHostName, portNumber)
-        # user
-        self.user = user
-        # password
-        self.password = password
 
-        self.connect(user, password)
+        self.conn = getEngineFromArgs(username=user, password=passwd, host=dbHostName, port=portNumber).connect()
 
     def loadSql(self, filePath, database):
         """Load an SQL file into a database
         @param filePath: the SQL file to load
         @param database: the database to use
         """
-        self.loadSqlScript(filePath, self.user, self.password, database)
+        utils.loadSqlScript(self.conn, filePath, database)
+
+    def execCommand0(self, cmd):
+        """Execute a command with no results
+        @param cmd SQL command to execute
+        """
+        self.conn.execute(cmd)
```

Comment:
I'd redefine execCommandX() methods to include additional arbitrary parameters and pass that to conn.execute(), something like this:

    def execCommand0(self, cmd, *args):
        self.conn.execute(cmd, *args)

The reason for that is the other piece of code that I looked at - `DbRecord.getInsertString()` method where you construct literal query by "manually" interpolating values into query string. You are not supposed to do that yourself, database API provides easier and more reliable way to do that via special expansion mechanism (http://docs.sqlalchemy.org/en/rel_1_1/core/connections.html#sqlalchemy.engine.Connection.execute). Mysql backend uses `%s` parameter style so what you need is to construct a query string which contains those markers instead of actual values and also pass actual values as extra arguments to `execute()`. Here is how I'd implement that method:

	class DbRecord(object):
		def getInsertQuery(self, tableName):
			members = [attr for attr in dir(self) if not callable(
				getattr(self, attr)) and not attr.startswith("__")]

			values = [getattr(self, mem) for mem in members]
			values = ["" if value is None else value for value in values]

			valuePlaceholders = ','.join(['%s']*len(members))  # string like "%s,%s,%s,%s,..."
			columns = ",".join(members)
			cmd = "INSERT INTO {} ({}) VALUES ({})".format(tableName, columns, valuePlaceholders)
			# return non-interpolated query and value list
			return cmd, values

And then use this as (e.g. in `LogIngestor.ingest()`):

                cmd, args = ent.getInsertQuery(self.submissionsTable)
                self.dbm.execCommand0(cmd, *args)

This would also remove explicit dependency on MySQLdb which you have now in `dbRecord` module.